### PR TITLE
Polish UI for importing ReadTracker files

### PR DESCRIPTION
**Pull request by christoffer**
Made some tweaks to the UI for importing ReadTracker files so that it doesn't suck completely, nor rely on external storage being available :)

### CODE REVIEW

**Comment by anthonymonori** [comment]
Great to see that now it is possible to import previously exported archives using the UI. Once I press export option, it prompts me to choose an application to save it with (e.g. the default share menu), but even if I don't select any of them (e.g. press back), it will still create a json archive inside the app's files folder. That is OK, although a bit confusing to the avg. user, I would say. I would still prefer some kind of feedback like a toast or a notification for example.

Once importing, I can't seem to navigate to where other apps would store these files. E.g. I tried exporting to my Dropbox share, but I can't seem to navigate to that folder using the explorer. Maybe missing permissions?

File: app/src/main/java/com/readtracker/android/activities/ImportActivity.java [hunk 1]
```diff
@@ -30,21 +35,23 @@
   private static final String TAG = ImportActivity.class.getSimpleName();
```
Comment (anthonymonori) [inline app/src/main/java/com/readtracker/android/activities/ImportActivity.java]: Small little note: in some classes you use ```class.getSimpleName()``` - like here - and in other you use ```class.getName()``` - like in the SettingsActivity f.x.

File: app/src/main/java/com/readtracker/android/activities/ImportActivity.java [hunk 2]
```diff
@@ -30,21 +35,23 @@
...
 
   @InjectView(R.id.current_folder_text) TextView currentFolderText;
+  @InjectView(R.id.button_common_dirs) TextView commonDirsButton;
   @InjectView(R.id.file_list) ListView fileList;
 
   private FileBrowserAdapter fileBrowseAdapter;
```
Comment (anthonymonori) [inline app/src/main/java/com/readtracker/android/activities/ImportActivity.java]: ```fileBrowseAdapter``` -> ```mFileBrowseAdapter``` to keep the naming conventions consistent?

File: app/src/main/java/com/readtracker/android/activities/ImportActivity.java [hunk 3]
```diff
@@ -53,9 +60,18 @@
...
 
+    mHomeDirFile = getFilesDir();
+    mDownloadsDirFile = Environment.getExternalStoragePublicDirectory(
+        Environment.DIRECTORY_DOWNLOADS
+    );
+    mSDCardFile = getExternalFilesDir(null);
+
+    registerForContextMenu(commonDirsButton);
+
     // NOTE Defaulting to downloads directory because that's the location most likely to contain
```
Comment (anthonymonori) [inline app/src/main/java/com/readtracker/android/activities/ImportActivity.java]: Note not relevant anymore?!

**Comment by christoffer** [comment]
> Great to see that now it is possible to import previously exported archives using the UI. Once I press export option, it prompts me to choose an application to save it with (e.g. the default share menu), but even if I don't select any of them (e.g. press back), it will still create a json archive inside the app's files folder. That is OK, although a bit confusing to the avg. user, I would say. I would still prefer some kind of feedback like a toast or a notification for example.

Ah, yeah fair point actually. I'll play around with having a little toast. Good suggestion, thanks!

> Once importing, I can't seem to navigate to where other apps would store these files. E.g. I tried exporting to my Dropbox share, but I can't seem to navigate to that folder using the explorer. Maybe missing permissions?

Do you mean you're exporting to the Dropbox app? In that case, it *should* be possible to navigate to the `/sdcard/Android/data/com.dropbox.android/files` dir (or equivalent on your phone), but that folder only holds the "offline" files for Dropbox, so you'll have to mark it as such in the Dropbox app.

File import/export in general is really clunky on Android when it's not images :/

**Comment by anthonymonori** [comment]
> Do you mean you're exporting to the Dropbox app? In that case, it should be possible to navigate to the /sdcard/Android/data/com.dropbox.android/files dir (or equivalent on your phone), but that folder only holds the "offline" files for Dropbox, so you'll have to mark it as such in the Dropbox app.

Once I press up, I see no other folders while navigating. Once I navigate outside of the application's folder ```/data/user/0/com.readtracker.debug/*```, then I can't go back and I am getting the following error message:
```Failed to open {foldername}. Maybe you don't have access to the file?``` Keep in mind I am testing in on a Nexus 6P running the latest official version of Android. Seems like some permission issues?!

> File import/export in general is really clunky on Android when it's not images :/

I agree, but does it makes sense to have such *hack-y* solution? Using the ```ShareActionProvider``` intent would allow them to choose an app to share it with, then the import could use a file picker dialog? What do you think? **My assumption** is that this app is more used by regular, everyday people (who might not be power users on Android) in contrast to people, developers, who would know how to navigate the file system and pull/push files to import/export. Then again, **this is just an assumption I made**, it might not be true and this solution could work fine as well. 

Q: Do you have any analytics on this app?

File: app/src/main/java/com/readtracker/android/activities/ImportActivity.java (hunk 2 above)
Comment (anthonymonori) [reply app/src/main/java/com/readtracker/android/activities/ImportActivity.java]: I've actually grown to dislike the Hungarian notation (even as a Hungarian 🤓), so I would like to retract my original (old) review comment on this

**Comment by anthonymonori** [comment]
I've gotten a crash when installed a fresh build from **this branch** on my phone and went directly to Settings and pressed Export

[huawei-nexus-6p-8.1.0-api-27-readtracker-import-readtracker-file-logcat.txt](https://github.com/christoffer/readtracker/files/1678983/huawei-nexus-6p-8.1.0-api-27-readtracker-import-readtracker-file-logcat.txt)

This works fine on master. Sorry if this is a WIP still. From what it looks like, [we need to implement FileProvider](https://developer.android.com/reference/android/support/v4/content/FileProvider.html) for this to work properly

**Comment by christoffer** [comment]
Ah, thanks for the bug report Antal! Yeah, this is still WIP. Ive
implemented The FileProvider, but Im going back and forth on exporting to
externalDir vs. sharing a temporary file using the send intent :)

**Comment by christoffer** [comment]
Closing this off for now. Looking into adding a library for the UI instead, since writing File browsing views in Android is terrible :)

**Pull request closed without merging**